app/main.py

```diff
@@ -1,8 +1,29 @@
+from __future__ import annotations
+
+
 class Person:
-    # write your code here
-    pass
+    people: dict = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        self.people[self.name] = self
+
+    @classmethod
+    def find_by_name(cls, name: str) -> Person | None:
+        return cls.people[name]
+
+
+def create_person_list(people: list) -> list[Person]:
+    for p in people:
```

Comment:
Such variable name as `p` is not allowed as stated by `flake8`. Replace it with a more descriptive one, e.g. `person`.